pkg/scheduler/cache/event_handlers.go

```diff
@@ -605,13 +605,34 @@ func (sc *SchedulerCache) SyncNode(nodeName string) error {
 		return err
 	}
 
+	if !sc.nodeCanAddCache(node) {
+		return nil
+	}
+	nodeCopy := node.DeepCopy()
 	csiNode, err := sc.csiNodeInformer.Lister().Get(nodeName)
 	if err == nil {
-		sc.setCSIResourceOnNode(csiNode, node)
+		sc.setCSIResourceOnNode(csiNode, nodeCopy)
 	} else if !errors.IsNotFound(err) {
 		return err
 	}
-	return sc.AddOrUpdateNode(node)
+	return sc.AddOrUpdateNode(nodeCopy)
+}
+
+func (sc *SchedulerCache) nodeCanAddCache(node *v1.Node) bool {
+	if !responsibleForNode(node.Name, sc.schedulerPodName, sc.c) {
+		return false
+	}
+	if len(sc.nodeSelectorLabels) == 0 {
+		return true
+	}
+	for labelName, labelValue := range node.Labels {
+		key := labelName + ":" + labelValue
```

Comment:
I have a small question: Can this `nodeSelectorLabels map[string]string` be adjusted to` map[string]struct{}`? In fact, it is just used as a set, and there is no need to use string to store variables.

Reply:
This is a small problem, it doesn't matter if it is not changed

Reply:
Yeah, it's an issues left over from history, but fix it is also ok.

Reply:
done here.